src/main/java/org/mitre/synthea/world/agents/Person.java

```diff
@@ -99,8 +102,18 @@
   public List<Module> currentModules;
   public Map<String, Object> attributes;
   public Map<VitalSign, ValueGenerator> vitalSigns;
-  Map<String, Map<String, Integer>> symptoms;
+  /** Data structure for storing symptoms faced by a person.
```

Comment:
I think all of the symptom-related Maps should be replaced by/coalesced into a new class or set of classes to make it clearer what data is being managed. Use simple bean classes where appropriate vs using generic collection classes to hold all the instance data. E.g. you might have a symptom class with properties for name, source, onset time, end time, last updated etc. The new methods added to Person that work with these Maps should be moved into the new class/classes to provide a clean encapsulation.

Reply:
Hi @hadleynet. I understand what you mean. I will refactor the code accordingly.